net/netcheck/netcheck.go

```diff
@@ -864,22 +870,29 @@ func (c *Client) GetReport(ctx context.Context, dm *tailcfg.DERPMap) (_ *Report,
 		return c.finishAndStoreReport(rs, dm), nil
 	}
 
-	ifState, err := interfaces.GetState()
-	if err != nil {
-		c.logf("[v1] interfaces: %v", err)
-		return nil, err
+	var ifState *interfaces.State
+	if c.LinkMon == nil || reflect.ValueOf(c.LinkMon).Kind() == reflect.Ptr && reflect.ValueOf(c.LinkMon).IsNil() {
```

Comment:
That's not what I would've expected from the docs on the Client.LinkMon type.

Also it means you can't implement `interfaces.LinkMonitor` with a nil pointer receiver. I don't see that being a documentation restriction. That's a valid thing to do in Go.

Reply:
This was mostly a bandaid because I was getting panics -- we must have a `var linkMon *monitor.Mon` somewhere or something like it. Is there a preferred pattern I should use instead?

Reply:
I'm forgetting the current package dep graph today vs what you're proposing vs what I was toying with the other day, but maybe something like adding a method on `monitor.(*Mon)` that returns a nil interface or a non-nil interface of itself?

```go
// Interface returns m as an interface, if m is non-nil.
// If nil, it returns a nil interface.
func (m *Mon) Interface() LinkMonitor {
    if m == nil {
        return nil
    }
    return m
}
```

(You'd have to duplicate the LinkMonitor definition in the `wgengine/monitor` package but that's okay... docs can cross-reference each other)

Reply:
It seems even nicer to not have the interface at all, and just have a pointer to a `Mon` directly. I sent out #7917 to make it easier to do that without the import path making it look like we're pulling in other wireguard bits.